packages/plugin-node-modules/sources/NodeModulesLinker.ts

```diff
@@ -514,20 +403,23 @@ const copyPromise = async (dstDir: PortablePath, srcDir: PortablePath, {baseFs}:
   }
 };
 
-const persistNodeModules = async (rootPath: PortablePath, prevLocatorMap: NodeModulesLocatorMap, locatorMap: NodeModulesLocatorMap, baseFs: FakeFS<PortablePath>, report: Report) => {
-  const rootNmDirPath = ppath.join(rootPath, NODE_MODULES);
-  const locatorStatePath = ppath.join(rootNmDirPath, LOCATOR_STATE_FILE);
+async function persistNodeModules(preinstallState: NodeModulesLocatorMap | null, installState: NodeModulesLocatorMap, {baseFs, project, report}: {project: Project, baseFs: FakeFS<PortablePath>, report: Report}) {
+  const rootNmDirPath = ppath.join(project.cwd, NODE_MODULES);
 
-  const prevLocationTree = buildLocationTree(prevLocatorMap);
-  const locationTree = buildLocationTree(locatorMap);
+  const prevLocationTree = buildLocationTree(preinstallState, {skipPrefix: project.cwd});
```

Comment:
What if project folder is physically moved to the other place on disk? Previously this would not result in invalidation of install state. I'm worried that with current code it will. I'm sure it should not invalidate install state.

Reply:
The state will still be saved on disk with relative paths ([here](https://github.com/yarnpkg/berry/pull/717/files/18a43a3a9967b02b5dcc498bca4aefb16d7e3377#diff-65485a2bfa64f0dae407c84cadb5db1cR176)), so even if the folder isn't at the same location from an install to another it'll work as before.

Reply:
Cool!